source/developers/concepts/hooks_extension_framework.rst

```diff
@@ -0,0 +1,155 @@
+|                    |  metadata associated with the event.                                   |  and the parameters it should receive.                      |
++--------------------+------------------------------------------------------------------------+-------------------------------------------------------------+
+| **Implementation** |  Implemented using `Django signals`_, which allow                      |  Implemented using an accumulative pipeline mechanism which |
+|                    |  developers to send and receive notifications that an action happened  |  takes a set of arguments and returns a modified set        |
+|                    |  within a Django application.                                          |  to the caller or raises exceptions during                  |
+|                    |                                                                        |  processing.                                                |
++--------------------+------------------------------------------------------------------------+-------------------------------------------------------------+
+| **Use cases**      |  Send an email notification when a user enrolls in a course.           |  Include additional information in an API endpoint response.|
+|                    |  an email notification.                                                |                                                             |
++--------------------+------------------------------------------------------------------------+-------------------------------------------------------------+
+
+.. _Django signals: https://docs.djangoproject.com/en/4.2/topics/signals/
+
+How to know when to use an Event or a Filter?
+----------------------------------------------
+
+When to use an Open edX Event?
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+Use an Open edX Event when but not limited to:
+
+- Trigger custom logic or processing in response to specific actions within the platform, e.g., updating a search index after a course block is modified.
+- Communicate, synchronize, or coordinate with other components or services based on specific events or actions, e.g., send certificate data from LMS to credentials service to keep models up to date.
+- Integrate with external systems or services based on specific events or actions within the platform, e.g., send user data to third-party services upon registration for marketing purposes.
+
+In summary, events can be used to integrate application components with each other or with external services, allowing them to communicate, synchronize, and perform additional actions when specific triggers occur.
+
+You can review the `Open edX Events`_ documentation for more information on `how to use events`_ in your project. This documentation includes a `list of available events`_ and `how to implement event receivers`_.
+
+.. _Open edX Events: https://docs.openedx.org/projects/openedx-events/en/latest/
```

Comment:
This is the second definition of this in the document. I suggest compiling your links at the very end of the document, rather than the end of each section, to avoid duplicates.

Reply:
Thank you, that'd be better indeed. You can review the changes here: Here you can review the changes: